packages/jet-brains-integration/src/web-types-generator.ts

```diff
@@ -71,33 +71,58 @@ function getJsProperties(
   };
 }
 
+/**
+ * @param field The Custom Elements Manifest class field to evaluate
+ * @return Whether the Custom Elements Manifest class field is a public property
+ */
+function isPublicProperty(field: schema.ClassField): boolean {
+  // It appears that JetBrains Web Types assumes that all properties are public (TS only) and not
+  // static.
+  return field.static !== true && (field.privacy === 'public' || field.privacy === undefined)
+}
+
 function getWebTypeProperties(
   component: Component,
-  typesSrc = "types"
-): WebTypeAttribute[] {
+  typesSrc?: string
+): WebTypeJsProperty[] {
   return (
-    ((component.attributes || component.members) as schema.Attribute[])?.map(
-      (attr) => {
+    (component.members?.filter((member) => member.kind === 'field') as schema.ClassField[])
+      .filter(isPublicProperty)
+      .map((field) => {
+        let type: string | undefined = field.type?.text;
+        if (typesSrc) {
+          if (typesSrc in field) {
+            type = (field as any)[typesSrc]?.text;
+          } else {
+            logYellow(
```

Comment:
I don't think we want to log this out for every property since there will be many that don't have custom types. Can we keep it simple with something like this?

`type: (attr as any)[typesSrc]?.text || attr.type?.text`

Reply:
Ah okay, I was assuming that the expanded types plugin would give every element an expanded type field and that users would want to know if it was missing during conversion. I'll make that update now.